backend/src/main/java/elbin_bank/issue_tracker/issue/domain/Issue.java

```diff
@@ -3,24 +3,60 @@
 import elbin_bank.issue_tracker.common.domain.BaseEntity;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
-import lombok.NoArgsConstructor;
-import lombok.Setter;
 import org.springframework.data.annotation.Id;
 import org.springframework.data.relational.core.mapping.Table;
 
 @Table("issue")
 @AllArgsConstructor
-@NoArgsConstructor
 @Getter
-@Setter
 public class Issue extends BaseEntity {
 
     @Id
-    private long id;
-    private long authorId;
-    private long milestoneId;
+    private final Long id;
+    private final long authorId;
     private String title;
     private String contents;
+    private Long milestoneId;
     private boolean isClosed;
 
+    public static Issue of(long id, long authorId, String title, String contents, Long milestoneId) {
+        return new Issue(id, authorId, title, contents, milestoneId, false);
+    }
+
+    public static Issue of(long authorId, String title, String contents, Long milestoneId, boolean isClosed) {
+        return new Issue(null, authorId, title, contents, milestoneId, isClosed);
+    }
+
+    public void changeState(boolean targetClosed) {
+        if (this.isClosed == targetClosed) {
+            return;
+        }
+
+        this.isClosed = targetClosed;
+    }
+
+    public void changeTitle(String title) {
+        if (this.title.equals(title)) {
+            return;
+        }
+
+        this.title = title;
+    }
```

Comment:
```suggestion
        this.title = title;
    }
```
변경 여부에 무관하게 업데이트 쿼리가 나가게된다면 분기를 줄이는게 좋지 않을까요 ㅎㅎ 이렇게해도 동일하게 동작할거같습니다.

Reply:
https://github.com/codesquad-members-2025/issue-tracker/pull/44#discussion_r2119962043
를 기반으로 동일한 경우 쿼리를 나가지 않게 처리해보겠습니다